netsim/modules/_routing.py

```diff
@@ -69,6 +69,29 @@ def routing_af(node: Box, proto: str, features: typing.Optional[Box] = None) ->
         log.IncorrectValue,
         proto)
 
+# After parsing/building the IGP AF structure an offloading VRF interfaces into VRF data
+# structure, recheck the interface status. If there's no match between interface AF and IGP AF,
+# remove IGP from the interface
+
+def check_interface_af(node: Box, proto: str) -> None:
+  for vrf_proto in routing_protocol_data(node,proto):
+    interfaces = vrf_proto.get('interfaces',node.interfaces)                # Get VRF IGP interfaces or global intf
+    if 'af' not in vrf_proto:                                               # Is this IGP instance limited by AFs?
+      continue
+    AF_mismatch = False
+    for intf in interfaces:
+      if proto not in intf:                                                 # IGP not active on this interface
+        continue
+      if any((intf.get(af,False) or intf.get(f'dhcp.client.{af}',False))
```

Comment:
We don't use it anywhere afaik, but note that ```intf.get(af)``` returns ```None``` if the key doesn't exist; the ```False``` could be omitted

Reply:
You're absolutely correct, thank you! However, IIRC, behavior changed across various versions of Box, so I learned (the hard way) to be on the safe side.